bindings/rust/s2n-tls/src/raw/config.rs

```diff
@@ -245,15 +245,55 @@ impl Builder {
         Ok(self)
     }
 
-    /// # Safety
+    pub fn wipe_trust_store(&mut self) -> Result<&mut Self, Error> {
+        unsafe { s2n_config_wipe_trust_store(self.as_mut_ptr()).into_result()? };
+        Ok(self)
+    }
+
+    /// Sets whether or not a client certificate should be required to complete the TLS connection.
+    /// See: https://github.com/aws/s2n-tls/blob/main/docs/USAGE-GUIDE.md#client-auth-related-calls
+    pub fn set_client_auth_type(
+        &mut self,
+        auth_type: s2n_cert_auth_type::Type,
+    ) -> Result<&mut Self, Error> {
+        unsafe { s2n_config_set_client_auth_type(self.as_mut_ptr(), auth_type).into_result() }?;
+        Ok(self)
+    }
+
+    /// Set a custom callback function which is run during client certificate validation during
+    /// a mutual TLS handshake.
     ///
-    /// The `context` pointer must live at least as long as the config
-    pub unsafe fn set_verify_host_callback(
+    /// The callback may be called more than once during certificate validation as each SAN on
+    /// the certificate will be checked.
+    pub fn set_verify_host_handler<T: 'static + VerifyClientCertificateHandler>(
         &mut self,
-        callback: s2n_verify_host_fn,
-        context: *mut core::ffi::c_void,
+        handler: T,
     ) -> Result<&mut Self, Error> {
-        s2n_config_set_verify_host_callback(self.as_mut_ptr(), callback, context).into_result()?;
+        unsafe extern "C" fn verify_host_cb(
+            host_name: *const ::libc::c_char,
+            _host_name_len: usize,
+            context: *mut ::libc::c_void,
+        ) -> u8 {
+            let maybe_cstr = CStr::from_ptr(host_name).to_str();
+            if let Ok(host_name_str) = maybe_cstr {
```

Comment:
it would be safer/faster to use the provided len rather than look for the `\0'

```suggestion
        unsafe extern "C" fn verify_host_cb(
            host_name: *const ::libc::c_char,
            host_name_len: usize,
            context: *mut ::libc::c_void,
        ) -> u8 {
            let host_name = host_name as *const u8;
            let host_name = core::slice::from_raw_parts(host_name, host_name_len);
            if let Ok(host_name_str) = core::str::from_utf8(host_name) {
```